src/bin/common.rs

```diff
@@ -294,11 +301,7 @@ pub fn process_frame(
       _ => { has_data = false; }
     }
   }
-  if read_frame {
-    Ok(frame_summaries)
-  } else {
-    Err(())
-  }
+  Ok(frame_summaries)
```

Comment:
Can this method still return an `Err`? If not, we should simplify the return type by removing the `Result` wrapper.